core/templates/domain/learner_group/learner-group.model.ts

```diff
@@ -0,0 +1,207 @@
+    this._subtopicPageIds = subtopicPageIds;
+    this._storyIds = storyIds;
+  }
+
+  get id(): string {
+    return this._id;
+  }
+
+  get title(): string {
+    return this._title;
+  }
+
+  set title(title: string) {
+    this._title = title;
+  }
+
+  get description(): string {
+    return this._description;
+  }
+
+  set description(description: string) {
+    this._description = description;
+  }
+
+  get facilitatorUsernames(): string[] {
+    return this._facilitatorUsernames;
+  }
+
+  get studentUsernames(): string[] {
+    return this._studentUsernames;
+  }
+
+  get invitedStudentUsernames(): string[] {
+    return this._invitedStudentUsernames;
+  }
+
+  get subtopicPageIds(): string[] {
+    return this._subtopicPageIds;
+  }
+
+  get storyIds(): string[] {
+    return this._storyIds;
+  }
+
+  addFacilitator(facilitatorUsername: string): void {
+    this._facilitatorUsernames.push(facilitatorUsername);
+  }
+
+  removeFacilitator(facilitatorUsername: string): void {
+    let index = this._facilitatorUsernames.indexOf(facilitatorUsername);
+    this._facilitatorUsernames.splice(index, 1);
+  }
+
+  addStudent(studentUsername: string): void {
+    this._studentUsernames.push(studentUsername);
+  }
+
+  removeStudent(studentUsername: string): void {
+    let index = this._studentUsernames.indexOf(studentUsername);
+    this._studentUsernames.splice(index, 1);
+  }
+
+  inviteStudent(studentUsername: string): void {
+    this._invitedStudentUsernames.push(studentUsername);
+  }
+
+  revokeInvitation(studentUsername: string): void {
+    let index = this._invitedStudentUsernames.indexOf(studentUsername);
+    this._invitedStudentUsernames.splice(index, 1);
+  }
+
+  addSubtopicPageId(subtopicPageId: string): void {
+    this._subtopicPageIds.push(subtopicPageId);
+  }
+
+  removeSubtopicPageId(subtopicPageId: string): void {
+    let index = this._subtopicPageIds.indexOf(subtopicPageId);
+    this._subtopicPageIds.splice(index, 1);
+  }
+
+  addStoryId(storyId: string): void {
+    this._storyIds.push(storyId);
+  }
+
+  removeStoryId(storyId: string): void {
+    let index = this._storyIds.indexOf(storyId);
+    this._storyIds.splice(index, 1);
+  }
+
+  validate(): string[] {
+    let issues = [];
+    if (this._title === '') {
+      issues.push(
+        'Learner Group title should not be empty.');
+    }
+    if (this._description === '') {
+      issues.push(
+        'Learner Group description should not be empty.');
+    }
+    if (this._facilitatorUsernames.length === 0) {
+      issues.push(
+        'Learner Group should have at least one facilitator.');
+    }
+    const commonUsernames = this.studentUsernames.filter(
+      username => this.invitedStudentUsernames.includes(username)
+    );
+    if (commonUsernames.length > 0) {
+      issues.push('Students can not be invited to join the same group again.');
+    }
+
+    return issues;
+  }
+
+  preCreationValidate(): string[] {
```

Comment:
I think both validation functions can be merged into a single function, with a flag input for the one additional check. Wdyt?

Reply:
hmm, yeah, merged them both. Thanks!